### PR TITLE
Patient Consultation: Route to Facility 🏥, Admission Date & Time 🕚, Rename variable: `verified_by` to `treating_physician`

### LINKED ISSUE #1618: Rename "Verified by" to "treating physician"
**Is your feature request related to a problem? Please describe.**
This issue is associated with https://github.com/coronasafe/care_fe/issues/6278 
The label will be renamed from "verified by" to "treating physician" in frontend. 

Keeping the renaming in mind the field in the backend needs to be updated accordingly. 
Please refer to https://github.com/coronasafe/care_fe/issues/6278 to get a better idea. 

**Describe the solution you'd like**
Rename the field to a more relevant name.

### LINKED ISSUE #1662: Consultation admission date: Allow time to be captured & clean dirty time for existing records
### Related to:

- https://github.com/coronasafe/care_fe/issues/6280

Currently, we have dirty time on the admission date even though we have not collected time from the user so far. Hence we'll need to clean existing records by resetting it to `00:00 HRS` and properly input time from the user.

image

### LINKED ISSUE #1671: Patient Consultation: Route to Facility
As part of https://github.com/coronasafe/care_fe/issues/6280:

## Consultation Status → Route to Facility

- [ ] Add field `route_to_facility` (choices: `OUTPATIENT`, `INTER_FACILITY_TRANSFER`, `INTRA_FACILITY_TRANSFER`)
- [ ] Route to Facility should be a required field.
- [ ] Deprecate old field: `consultation_status` to `deprecated_consultation_status`
- [ ] Map `consultation_status` to `route_to_facility` as follows:

```yaml
Brought Dead:                    Outpatient/Emergency Room
Transferred from Ward:           Internal Transfer within the facility
Transferred from ICU:            Internal Transfer within the facility
Referred from another hospital:  Referred from another facility
Outpatient:                      Outpatient/Emergency Room
```

### Related places to be updated

- [ ] Update CSV Export
- [ ] Update Discharge Summary

### Fields for "Referred from another facility"

- [ ] Add field: `referred_from_facility`, `referred_from_external` facility (required)
- [ ] Add field: `referred_by_external_facility` user (textfield, non-mandatory)
- [ ] Add field: `icu_admission_date` (non-mandatory)

### Fields for "Internal Transfer within the Facility"
- [ ] Add field: `transferred_from_location` location field (required)
- [ ] Add field: `icu_admission_date` (non-mandatory)

### PR DESCRIPTION
**Pull request by rithviknishad**
## Associated Issue

- Issue https://github.com/coronasafe/care_fe/issues/6280
- Fixes #1671 
- Fixes #1662
- Fixes #1618
- FE PR https://github.com/coronasafe/care_fe/pull/6458

## Proposed Changes

- Fields added and renamed (refer table below)
- ⚠️ **Custom Migration**: reset time to 00:00 of current timezone (IST) to clean dirty time captured so far by front-end's Date only picker. (Front-end now captures both date and time from the user)
- ⚠️ **Custom Migration**: Populate `route_to_facility` based on deprecated `consultation_status` field value.
- Updated CSV Export and Discharge Summary to show Route to Facility instead of Consultation Status

### Renamed Fields
| **Old Field**         |           **New Field**          |                 **Reason**                 |
|-----------------------|:--------------------------------:|:------------------------------------------:|
| `consultation_status` | `route_to_facility` | Alter field + Align with new choices |
| `verified_by`         |       `treating_physician`       |     Correction (align with field label)    |

### New Fields

| **Field**                         |           **Type**          | **Validation** |                                              **Remarks**                                              |
|-----------------------------------|:---------------------------:|:--------------:|:-----------------------------------------------------------------------------------------------------:|
| `transferred_from_location`       | ForeignKey: `AssetLocation` |    required*   |                             *applicable only for `INTRA_FACILITY_TRANSFER`                            |
| `referred_from_facility`          |    ForeignKey: `Facility`   |  required* **  | *applicable only for `INTER_FACILITY_TRANSFER`; **XOR required with `referred_from_facility_external` |
| `referred_from_facility_external` |           VarChar           | required* **   | *applicable only for `INTER_FACILITY_TRANSFER`; **XOR required with `referred_from_facility`          |
| `referred_by_external`            |           VarChar           |                | applicable only for `INTER_FACILITY_TRANSFER`                                                         |
| `icu_admission_date`              |        DateTimeField        |                | NOT applicable for `OUTPATIENT/ER`                                                                    |

#### Admission Date reset to 00:00 of timezone - Before Migration

image

#### Admission Date reset to 00:00 of timezone - After Migration

image

#### Populate Route to Facility - Before Migration

```yaml
UNKNOW: 2587                      # mapped to null
BROUGHT_DEAD: 29                  # mapped to 10 (OUTPATIENT)
TRANSFERRED_FROM_WARD: 61         # mapped to 30 (INTRA_FACILITY_TRANSFER)
TRANSFERRED_FROM_ICU: 25          # mapped to 30 (INTRA_FACILITY_TRANSFER)
REFERRED_FROM_OTHER_HOSPITAL: 21  # mapped to 20 (INTER_FACILITY_TRANSFER)
OUT_PATIENT: 397                  # mapped to 10 (OUTPATIENT)
```

image

#### Populate Route to Facility - After Migration

```yaml
OUTPATIENT: 426              # expected 426 (OP 397 + BROUGHT_DEAD 29)
INTER_FACILITY_TRANSFER: 21  # expected 21 (REFERRED_FROM_OTHER_HOSPITAL 21)
INTRA_FACILITY_TRANSFER: 86  # expected 86 (WARD 61 + ICU 25)
```

image

### Architecture changes

- Remove this section if not used

## Merge Checklist

- [x] Tests added/fixed
- [ ] Update docs in `/docs`
- [x] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Comment by vigneshhari** [comment]
Tests failing

**Review by Ashesh3 — APPROVED** [review]

**Review by sainak — APPROVED** [review]

**Comment by vigneshhari** [comment]
This PR is probably causing me to age a bit faster :/

File: care/facility/migrations/0395_alter_patientconsultation_route_to_facility.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        qs = PatientConsultation.objects.all()
+
+        # Unknown -> None
+        qs.filter(route_to_facility=0).update(route_to_facility=None)
+        # Brought Dead/Outpatient -> Outpatient/Emergency Room
+        qs.filter(route_to_facility=1).update(route_to_facility=10)
+        qs.filter(route_to_facility=5).update(route_to_facility=10)
+        # Transferred from Ward/ICU -> Internal Transfer within facility
+        qs.filter(route_to_facility=2).update(route_to_facility=30)
+        qs.filter(route_to_facility=3).update(route_to_facility=30)
```
Comment (sainak) [inline care/facility/migrations/0395_alter_patientconsultation_route_to_facility.py:67]: nit:

```suggestion
        qs.filter(models.Q(route_to_facility=1) | models.Q(route_to_facility=5)).update(route_to_facility=10)
        # Transferred from Ward/ICU -> Internal Transfer within facility
        qs.filter(models.Q(route_to_facility=2) | models.Q(route_to_facility=3)).update(route_to_facility=30)
```

File: care/facility/models/patient_base.py [hunk 2]
```diff
@@ -106,16 +109,12 @@ class DiseaseStatusEnum(enum.IntEnum):
...
-    BROUGHT_DEAD = 1
-    TRANSFERRED_FROM_WARD = 2
-    TRANSFERRED_FROM_ICU = 3
-    REFERRED_FROM_OTHER_HOSPITAL = 4
-    OUT_PATIENT = 5
-
+class RouteToFacility(IntegerChoices):
+    OUTPATIENT = 10, _("Outpatient/Emergency Room")
+    INTER_FACILITY_TRANSFER = 20, _("Referred from another facility")
+    INTRA_FACILITY_TRANSFER = 30, _("Internal Transfer within the facility")
```
Comment (sainak) [inline care/facility/models/patient_base.py:116]:
```suggestion
```

**Review by sainak — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**